### PR TITLE
Add wavedrom module to library reference.

### PR DESCRIPTION
**Pull request by cmarqu**

### CODE REVIEW

File: documentation/source/library_reference.rst [hunk 1]
```diff
@@ -269,3 +269,14 @@ XGMII
     :members:
     :member-order: bysource
     :show-inheritance:
+
+Miscellaneous
+=============
+
+WaveDrom
```
Comment (imphil) [inline documentation/source/library_reference.rst]: Please call this already "A signal tracer for WaveDrom". The section title otherwise sounds like we provide wavedrom here, which we don't.

File: cocotb/wavedrom.py [hunk 2]
```diff
@@ -124,19 +115,17 @@ def get(self, add_clock=True):
...
-
-    Arguments are an arbitrary number of signals or busses to trace.
+    """Context manager to enable tracing of signals.
 
-    We also require a clock to sample on, passed in as a keyword argument
+    Arguments are an arbitrary number of signals or buses to trace.
+    We also require a clock to sample on, passed in as a keyword argument.
 
-    Usage:
+    Usage::
```
Comment (imphil) [inline cocotb/wavedrom.py]: Is that change intended?

File: documentation/source/library_reference.rst (hunk 1 above)
Comment (cmarqu) [reply documentation/source/library_reference.rst]: Yes, it is intended, with this the following code will be shown as code properly.

Comment (cmarqu) [reply documentation/source/library_reference.rst]: Argh, replied to the wrong comment. I will change the heading of course.

File: documentation/source/library_reference.rst [hunk 3]
```diff
@@ -269,3 +269,14 @@ XGMII
...
+--------
+
+.. automodule:: cocotb.wavedrom
+    :members:
+    :member-order: bysource
+    :synopsis: A signal tracer for WaveDrom.
```
Comment (eric-wieser) [inline documentation/source/library_reference.rst]: How does this piece render?

Comment (cmarqu) [reply documentation/source/library_reference.rst]: Like this: https://cocotb-cmarqu.readthedocs.io/en/add-wavedrom-to-lib-ref/library_reference.html#miscellaneous

Comment (eric-wieser) [reply documentation/source/library_reference.rst]: So, this line does not render at all?

Comment (cmarqu) [reply documentation/source/library_reference.rst]: Ah, the synopsis line you mean... This is needed and apparently only used for the Module Index: https://cocotb-cmarqu.readthedocs.io/en/add-wavedrom-to-lib-ref/py-modindex.html

File: documentation/source/library_reference.rst (hunk 1 above)
Comment (cmarqu) [reply documentation/source/library_reference.rst]: Oh, I had forgotten about that... I'll be without access to a computer until the end of the next week unfortunately.
